### DISCUSSION TITLE
JDK native threads cannot join link

### DISCUSSION THREAD

**Discussion by HTFX-Cloud** [apache/skywalking#10385] [Q&A]
-Currently Spring's @async, @TraceCrossThread, and RunnableWrapper are all supported for asynchronization, but the code needs to be revamped. There are a large number of native Thread.start writings in the code, as follows

Thread thread = new Thread(() -> {

RestTemplate restTemplate = new RestTemplate();

String body = restTemplate.getForEntity (Request Address, String.class). getBody();

};

Thread.start();

The logic of this notation cannot be joined to the link at this time

-Add apm-jdk-threading-plugin plugin application and SW_in environment variable PLUGIN_ JDKTHREADING_ THREADING_ CLASS_ PREFIXES specifies java.lang.Thread but native threading still cannot be added (currently the plug-in source code has enhancements to java.lang.runnable)

-For apm-jdk-threading-plugin plugins, required SW_ PLUGIN_ JDKTHREADING_ THREADING_ CLASS_ How PREFIXES is used (granularity of package names, what packages can and cannot be added, etc.)

**Comment by wu-sheng** [comment]
`Thread` class is used in the agent, so we are not going to intercept it. This would be a kind of Game Theory.
Meanwhile, instrument `Thread` could easily cause unexpected crashing and memory leak in a complex application.

Considering both ways, you will not find this kind of thing in SkyWalking.

**Reply by HTFX-Cloud** [reply]
Therefore, the programming method using native threads can only be modified, such as @ async, @ TraceCrossThread, or RunnableWrapper?

**Reply by HTFX-Cloud** [reply]
Another problem is the SW used in the apm-jdk-threading-plugin plug-in_ PLUGIN_ JDKTHREADING_ THREADING_ CLASS_ What scenarios should the PREFIXES keyword be used in?
